great_expectations_cloud/agent/actions/run_checkpoint.py

```diff
@@ -15,10 +15,16 @@
 class RunCheckpointAction(AgentAction[RunCheckpointEvent]):
     @override
     def run(self, event: RunCheckpointEvent, id: str) -> ActionResult:
+        # TODO: move connection testing into OSS; there isn't really a reason it can't be done there
+        for datasource_name in event.datasource_names:
+            datasource = self._context.get_datasource(datasource_name)
+            datasource.test_connection(test_assets=True)  # raises `TestConnectionError` on failure
```

Comment:
I would suggest only testing the connection for the asset that the check is being run against.
A problem with asset `A` shouldn't affect my ability to validate asset `B`.

Every asset type also has a `test_connection()` method that can called.
```python
# something like this
        for datasource_name in event.datasource_names:
            datasource = self._context.get_datasource(datasource_name)
            datasource.test_connection(test_assets=False)
            for asset_name in checkpoint_assets:
                asset = datasource.get_asset(asset_name)
                asset.test_connection()
```

Reply:
makes sense--thanks for the explanation; will update!

Reply:
ok, this PR has been updated - I also updated the [corresponding mercury PR](https://github.com/greatexpectationslabs/mercury/pull/2772)